ast/src/parsed/mod.rs

```diff
@@ -465,6 +525,7 @@ pub enum Expression<Ref = NamespacedPolynomialReference> {
     MatchExpression(SourceRef, MatchExpression<Self>),
     IfExpression(SourceRef, IfExpression<Self>),
     BlockExpression(SourceRef, BlockExpression<Self>),
+    StructExpression(SourceRef, StructExpression<Self>),
```

Comment:
concerning https://github.com/powdr-labs/powdr/pull/1827/files#r1775775696 - it might work to use `StructExpression(SourceRef, StructExpression<Ref>)` here (and then use `fields: Vec<NamedExpression<Box<Expression<Ref>>>>` or something below).